aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -1793,17 +1793,17 @@ Tensor& var_out(const Tensor& self, IntArrayRef dim, bool unbiased, bool keepdim
 
 Tensor std(const Tensor& self, bool unbiased) {
   return at::std(
-      self, /*dim=*/c10::nullopt, /*correction=*/int64_t{unbiased ? 1 : 0});
+      self, /*dim=*/c10::nullopt, /*correction=*/c10::make_optional<int64_t>({unbiased ? 1 : 0}));
 }
 
-Tensor std(const Tensor& self, IntArrayRef dim, bool unbiased, bool keepdim) {
-  return at::std(self, /*dim=*/at::OptionalIntArrayRef(dim),
-                 /*correction=*/int64_t{unbiased ? 1 : 0}, keepdim);
+Tensor std(const Tensor& self, at::OptionalIntArrayRef dim, bool unbiased, bool keepdim) {
+  return at::std(self, dim,
+                 /*correction=*/c10::make_optional<int64_t>({unbiased ? 1 : 0}), keepdim);
```

Comment:
Because of the similarity between the signatures of `std.dim` and `std.correction`, these `make_optional`s are required to call the correct overload of `at::std()`. I'm not sure if there's a way to avoid this BC break for the C++ API

It seems like combining `std.dim` and `std.correction` would make things simpler, but that would also be BC breaking I think